core/storage/app_feedback_report/gae_models.py

```diff
@@ -0,0 +1,659 @@
+            if not cls.get_by_id(new_id):
+                return new_id
+        raise Exception(
+            'The id generator for AppFeedbackReportModel is producing too '
+            'many collisions.')
+
+    @classmethod
+    def scrub_report(cls, report_id, scrubbed_by):
+        """Scrubs the instance of AppFeedbackReportModel with given ID, removing
+        any user-entered input in the entity.
+
+        Args:
+            report_id: str. The id of the model entity to scrub.
+            scrubbed_by: str. The id of the user or cron job that is intiating
+                scrubbing this report.
+        """
+        cls._scrub_report_in_transaction(report_id, scrubbed_by)
+
+    @classmethod
+    @transaction_services.run_in_transaction_wrapper
+    def _scrub_report_in_transaction(cls, report_id, scrubbed_by):
+        """See scrub_report for general documentaion of what this method does.
+        It's only safe to call this method from within a transaction.
+
+        Args:
+            report_id: str. The id of the model entity to scrub.
+            scrubbed_by: str. The id of the user or cron job that is intiating
+                scrubbing this report.
+        """
+        report_entity = cls.get_by_id(report_id)
+        if not report_entity:
+            raise Exception(
+                'The AppFeedbackReportModel trying to be scrubbed does not '
+                'exist.')
+        if report_entity.platform == PLATFORM_CHOICE_ANDROID:
+            scrubbed_report_info = cls._scrub_report_info(
+                report_entity.android_report_info)
+            report_entity.android_report_info = scrubbed_report_info
+        else:
+            scrubbed_report_info = cls._scrub_report_info(
+                report_entity.web_report_info)
+            report_entity.web_report_info = scrubbed_report_info
+        report_entity.scrubbed_by = scrubbed_by
+        report_entity.update_timestamps()
+        report_entity.put()
+
+    @staticmethod
+    def _scrub_report_info(report_info_dict):
+        """Scrubs the dictionary of any fields that contains input directly from
+        the user.
+
+        Args: report_info_dict: dict. The info dict collected in a report.
+
+        Returns: dict. The scrubbed report.
+        """
+        new_report_info = dict()
+        for key in report_info_dict:
+            if not REPORT_INFO_TO_REDACT.__contains__(key):
+                new_report_info[key] = report_info_dict[key]
+        return new_report_info
+
+    @staticmethod
+    def get_deletion_policy():
+        """Model stores the user ID of who has scrubbed this report for auditing
+        purposes but otherwise does not contain data directly corresponding to
+        the user themselves.
+        """
+        return base_models.DELETION_POLICY.LOCALLY_PSEUDONYMIZE
+
+    @classmethod
+    def get_export_policy(cls):
+        """Model contains data referencing user and will be exported."""
```

Comment:
@BenHenning I feel like this is over-exporting. Why would the scrubber need all this information about the report?

Reply:
I was thinking about this too, and I think I agree that this is over-exporting, especially since this isn't really related to the scrubber themselves. I'm thinking just the platform, platform_version, ticket_id, submitted_on, report_type, and category is enough, WDYT?

Reply:
Can you check with @BenHenning? I think he decides. I think what you have seems fine though it might still be on the side of too much information.

Reply:
I agree with the over-exporting. We really only need sufficient data to provide context to the affected user (the person who scrubbed the report). I think the data you suggested is sufficient to build that context @jcqli.

Reply:
To clarify: I agree that this was previously over-exporting* and that we should reduce it. :)

Reply:
Done, reduced it to the above listed fields.